pkg/lightweight/lightweight.go

```diff
@@ -184,21 +184,38 @@ func find(name string, origin string) map[string]*PartBag {
 
 			versionSort(versions)
 
-			// if oems were removable, there'd be know way of
+			typ := s.typ
+			inst := s.inst
+
+			// if oems were removable, there'd be no way of
 			// telling the kind of a removed origin-less package
-			if s.typ == pkg.TypeFramework && helpers.FileExists(filepath.Join(dirs.SnapOemDir, name)) {
-				s.typ = pkg.TypeOem
-				s.inst = dirs.SnapOemDir
+			//
+			// in case it's not clear, we're walking the *data*
+			// directory, where directories for packages of all
+			// types are present. OEM packages and frameworks look
+			// the same in the data dir: they both have no
+			// origin. However, OEM packages are uninstallable, and
+			// you can't install a package with the same name as an
+			// active package, so if /oem/{name} exists, we switch
```

Comment:
It feels slightly frightening to offer authority based on whether it has the right name, even more considering we have plans of introducing renames into the system.

That said, this is a fix rather than a redesign, and the comment is just pointing out what is already in place, so LGTM until we have a better plan.